src/core/common/logging.hpp

```diff
@@ -1146,6 +1146,40 @@ extern "C" {
  */
 void otDump(otLogLevel aLevel, otLogRegion aRegion, const char *aId, const void *aBuf, const size_t aLength);
 
+#ifdef OPENTHREAD_CONFIG_LOG_PREPEND_REGION
+/**
+ * This method converts the log region value into a string
+ *
+ * @param[in]  aRegion  The log region.
+ *
+ * @returns A const char pointer to the C string corresponding to the log region.
+ *
+ */
+const char *otLogRegionToString(otLogRegion aRegion);
+#endif // OPENTHREAD_CONFIG_LOG_PREPEND_REGION
+
+/**
+ * Local/private macro to format the log message
+ */
+#ifdef OPENTHREAD_CONFIG_LOG_PREPEND_REGION
```

Comment:
Perhaps this ifdef and the endif right above it are not needed, right? (so lines 1159 and 1164 could be removed)